### ISSUE TITLE
[User flows and navigating] User can't review, edit, or confirm transactional information before submission.  (03.09.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#59491]
#### VFS team name
MHV

#### VFS product name
My HealtheVet

#### VFS feature name
Secure Messaging

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
User can't review, edit, or confirm transactional information before submission. 

### Issue Details
On [/compose/](https://staging.va.gov/my-health/secure-messages/compose), when deleting an attached file, the Remove button is a single-click action with no step to confirm the action. Additionally, there's nothing announced by screen readers to indicate that a change has occurred --- focus just moves to the next focusable element.

It's easy to imagine some scenarios where a user might accidentally trigger the Remove button and not realize that they did it. Screen reader users may hit the wrong key but think they hit [Tab]. A low-vision user on a touch device might pinch-to-zoom and accidentally tap the Remove button in the process, and not see the issue disappear.

I'm flagging this as an issue with experience standard 03.09, which maps to WCAG 3.3.4. It's not an exact match but I think a situation like this is largely covered under 3.3.4.

### VA.gov Experience Standard
[Category Number 03, Issue Number 09](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 3.3.4_AA

### Platform Recommendation
Two recommended changes:

1. Some kind of status message that receives focus when an issue has been removed, similar to the background-only alerts in use when changing contact information. If a visible status message isn't practical, at least some aria-live update should be announced to screen reader users.
2. Some kind of intermediate confirmation step between a user clicking Delete and the issue actually being removed. I believe the team working on the Supplemental Claims Form 20-0995 is exploring using a modal and intends to contribute back to the pattern in the design system. The confirmation modal for the "Delete draft" button (deleting the entire message draft) is a great model to follow.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by bryan-riley-va** [comment]
JIRA ticket created: https://jira.devops.va.gov/browse/MHV-45980

**Comment by coultonbunney-usds** [comment]
@artsymartha68 with Angela out for a few days, would you be able to help @vsaleem identify someone from CAIA who can approve the solution here?

**Comment by artsymartha68** [comment]
Tagging @sara-amanda on this

**Comment by sara-amanda** [comment]
## CAIA Inquiry
@coultonbunney-usds do you know if or when the changes will be merged in Staging? I ask, because I checked and it appears there is no notification in VO, at this time, that the attachment was removed. I can check back later, if the merge hasn't taken place yet.

**Comment by vsaleem** [comment]
> ## CAIA Inquiry
> @coultonbunney-usds do you know if or when the changes will be merged in Staging? I ask, because I checked and it appears there is no notification in VO, at this time, that the attachment was removed. I can check back later, if the merge hasn't taken place yet.

Hi Sara - I am working to get these changes in staging now. I'm waiting on one more approval before it can be merged.

**Comment by sara-amanda** [comment]
> Hi Sara - I am working to get these changes in staging now. I'm waiting on one more approval before it can be merged.

Sounds good @vsaleem ! Thanks!

**Comment by artsymartha68** [comment]
I said this in slack: Thank you Vic! We have a lot of folks OOO for the holiday at this point. We’ll fit it in as soon as we can. I'm not sure when @SarahKay8 will be back to take a look next week while Sara is OOO but we will get to this as soon as we can!

**Comment by SarahKay8** [comment]
HI @vsaleem! Are changes merged? If so may I have an updated link?

**Comment by SarahKay8** [comment]
Modal notification was great here, thanks for you work on this @vsaleem! https://images.zenhubusercontent.com/133843125/4aa39529-af1a-4149-b799-dda8a9654244/testing2.mp4

**Comment by SarahKay8** [comment]
Modal notification was great here, thanks for you work on this @vsaleem. Closing @artsymartha68 @sara-amanda 

https://images.zenhubusercontent.com/133843125/ed987d52-be32-420c-82d0-ae6d432edfd7/20230703173047951.mp4

**Comment by SarahKay8** [comment]
Hi @shiragoodman. Can you close this out? Having some trouble closing for some reason.
